apps/examples/testcase/le_tc/kernel/Kconfig

```diff
@@ -27,7 +27,7 @@ config TC_KN_ALL
 	select TC_KERNEL_LIBC_LIBGEN
 	select TC_KERNEL_LIBC_MATH if LIBM
 	select TC_KERNEL_LIBC_MISC
-	select TC_KERNEL_LIBC_MQUEUE
+	select TC_KERNEL_LIBC_MQUEUE if !BUILD_PROTECTED
```

Comment:
Please add dependency on DISABLE_MQUEUE as well.

Reply:
I think DISABLE_MQUEUE dependency is wrongly mentioned in TC_KERNEL_LIBC_QUEUE instead of TC_KERNEL_LIBC_MQUEUE . So I will rectified it in another PR.

Reply:
DISABLE_MQUEUE is disabling mqueue functionalities in kernel. LIBC_MQUEUE is a part of libc, not kernel so that adding dependency on DISABLE_MQUEUE is wrong.

Reply:
I checked libc mqueue. But it has a conditional as @yashwanth686007 said. We need to check it more.

Reply:
I found a message of Nuttx.
```
	* POSIX message queues:  Move mq_setattr() and mq_getattr() from
	  nuttx/libc/mqueue to nuttx/sched/mqueue.  Also add syscall support
	  for mq_setattr() and mq_getattr().  This is necessary in protected and
	  kernel builds because in those cases the message queue structure is
	  protected and cannot be accessed directly from user mode code (2015-06-03).
```
That's why it has DISABLE_MQUEUE conditional. We also need to move it to os/kernel/mqueue.

Reply:
But if we move libc mqueue to kernel, we don't need to merge this.
It is not difficult to do it so that how about closing this and moving it under kernel without BUILD_PROTECTED limitation?

Reply:
Ok we will move libc_mqueue tc's under os/kernel/mqueue. 

This PR has Timer tc also so I kept timer tc as it is and do libc_mqueue tc changes in another PR.
Is this way is fine??

Reply:
@NilabhShant Yeap. It looks good.
1. move mq_get/setattr files from lib/libc/mqueue to os/kernel/mqueue without BUILD_PROTECTED
2. move testcases of them to tc_mqueue.c and remove tc_libc_mqueue.c
3. leave only timer change in this commit

Reply:
Ok Sure.